Events Docs (#505)
tinacms/tinacms.org: content/docs/cms/events.md

---
title: Events
prev: /docs/cms/apis
next: /docs/cms/alerts
---

> The **Events** feature allows decouple parts of the CMS to notify the rest of the system about certain events which have happened.

Reviewer: ```suggestion
The **Events** feature allows decoupled parts of the CMS to notify the rest of the system about certain events which have happened.
```

> The **Events** feature allows decoupled parts of the CMS to notify the rest of the system about certain events which have happened.

Reviewer 2: ```suggestion
The **Events** feature allows decoupled parts of the CMS to notify the rest of the system that something has happened.
```

Reviewer 2: This sentence is kinda heavy to me. I'm not sure if my suggestion makes it better :man_shrugging:

Reviewer 2: ```suggestion
The **Events** feature allows decoupled parts of the CMS to:

  1. Notify the rest of the system that something has happened, and
  2. React to things that happen elsewhere in the system.
```

## Interface

```ts
interface Events {
  dispatch(event: Event): void

>   subscribe(event: string, listener: Listener): Unsubscribe

Reviewer 2: ```suggestion
  subscribe(eventType: string, listener: Listener): Unsubscribe
```

seeing the two `event` params next to each other with different types makes me think it would be helpful to call this `eventType`
